pennylane/optimize/adaptive.py

```diff
@@ -184,7 +184,7 @@ def step_and_cost(self, circuit, operator_pool, drain_pool=False, params_zero=Tr
             to the step, and the largest gradient
         """
         cost = circuit()
```

Comment:
@rmoyard Should we (can we) hard code `circuit.interface = 'autograd'` to prevent [this](https://github.com/PennyLaneAI/pennylane/blob/023e001cd437e5283a4ffa1b2f73bda38ebf4bc9/pennylane/qnode.py#L816) warning? Maybe we can just silent the warning here internally?

Reply:
Based on an offline chat, we decided to not hard-code `circuit.interface = 'autograd'` and leave it for the user to redefine their qnode when they see the warning.